eitprocessing/binreader/sequence.py

```diff
@@ -354,9 +380,33 @@ def select_by_time(
         else:
             end_index = bisect.bisect_left(self.time, end) - 1
 
-        return self[start_index:end_index]
+        return self.select_by_index(slice(start_index,end_index), label = label)
 
-    deepcopy = copy.deepcopy
+
+    def deepcopy(
```

Comment:
Wouldn't it be better to create the magic method `__deepcopy__()`, which will be used by `copy.deepcopy()`? In the current implementation, `copy.deepcopy(my_sequence)` would give a different result from `my_sequence.deepcopy()`. 

Writing your own `__deepcopy__()` methods is a bit confusing because you can't use `copy.deepcopy()` inside of them (resursion), but it solves the problem stated above.

Some considerations/suggestions:
- https://stackoverflow.com/questions/1500718/how-to-override-the-copy-deepcopy-operations-for-a-python-object
- https://stackoverflow.com/questions/15684881/python-implementation-of-shallow-and-deep-copy-constructors

Reply:
How about calling the current function `duplicate` (or similar) and then doing this (analogous to what I did for `__getitem__`:

```
    def __deepcopy__(self):
        return self.duplicate()
```

Reply:
That would work for me. Using `__deepcopy__ = duplicate` after the definition of `duplicate()` does the same (`__deepcopy__()` is then just an alias referring to the same method).

The problem is, however, that the function you wrote uses `copy.deepcopy()`. When calling `seq.duplicate()`, that function will call `copy.deepcopy(seq)`, which in turn will call `seq.duplicate()`. `Sequence.duplicate()` would have to be rewritten for this to work.

Reply:
I will look into this. I think it would be OK to merge this PR, and make a new PR to make sure that `copy.deepcopy(sequence)` and `sequence.deepcopy()` do the same. Maybe this could be done at the same time as working on #103, since they will both require rewriting copy implementations?

Reply:
So not even change it to `__deepcopy__`?

In that case, can you "formally" approve the PR, so that I can merge :)